src/System.Text.Json/tests/Serialization/OptionsTests.cs

```diff
@@ -321,5 +322,126 @@ public static void JsonEncodedTextStringsCustomAllowAll(string message, string e
+            GenericConverterTestHelper<short>("JsonConverterInt16", (short)5, "5");
+            GenericConverterTestHelper<int>("JsonConverterInt32", -100, "-100");
+            GenericConverterTestHelper<long>("JsonConverterInt64", (long)11111, "11111");
+            GenericConverterTestHelper<sbyte>("JsonConverterSByte", (sbyte)-121, "-121");
+            GenericConverterTestHelper<float>("JsonConverterSingle", 3.14f, "3.14");
+            GenericConverterTestHelper<string>("JsonConverterString", "Hello", "\"Hello\"");
+            GenericConverterTestHelper<ushort>("JsonConverterUInt16", (ushort)1206, "1206");
+            GenericConverterTestHelper<uint>("JsonConverterUInt32", (uint)3333, "3333");
+            GenericConverterTestHelper<ulong>("JsonConverterUInt64", (ulong)44444, "44444");
+            GenericConverterTestHelper<decimal>("JsonConverterDecimal", 3.3m, "3.3");
+            GenericConverterTestHelper<byte[]>("JsonConverterByteArray", new byte[] { 1, 2, 3, 4 }, "\"AQIDBA==\"");
+            GenericConverterTestHelper<DateTime>("JsonConverterDateTime", new DateTime(2018, 12, 3), "\"2018-12-03T00:00:00\"");
+            GenericConverterTestHelper<DateTimeOffset>("JsonConverterDateTimeOffset", new DateTimeOffset(new DateTime(2018, 12, 3, 00, 00, 00, DateTimeKind.Utc)), "\"2018-12-03T00:00:00+00:00\"");
+            Guid testGuid = new Guid();
+            GenericConverterTestHelper<Guid>("JsonConverterGuid", testGuid, $"\"{testGuid.ToString()}\"");
+            GenericConverterTestHelper<KeyValuePair<string, string>>("JsonKeyValuePairConverter`2", KeyValuePair.Create<string, string>("key", "value"), @"{""Key"":""key"",""Value"":""value""}");
+            GenericConverterTestHelper<Uri>("JsonConverterUri", new Uri("http://test.com"), "\"http://test.com\"");
+            GenericConverterTestHelper<Dictionary<string, long>>("DictionaryConverter", new Dictionary<string, long> { { "val1", 123 }, { "val2", 456 } }, "{\"val1\":103,\"val2\":436}");
+            GenericConverterTestHelper<long[]>("LongArrayConverter", new long[] { 1, 2, 3, 4 }, "\"1,2,3,4\"");
+        }
+
+        private static void GenericConverterTestHelper<T>(string converterName, object objectValue, string stringValue)
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new CustomConverterTests.LongArrayConverter());
+            options.Converters.Add(new CustomConverterTests.DictionaryConverter(20));
+
+            JsonConverter<T> converter = (JsonConverter<T>)options.GetConverter(typeof(T));
+
+            Assert.True(converter.CanConvert(typeof(T)));
+            Assert.Equal(converterName, converter.GetType().Name);
+
+            ReadOnlySpan<byte> data = Encoding.UTF8.GetBytes(stringValue);
+            Utf8JsonReader reader = new Utf8JsonReader(data);
+            reader.Read();
+
+            T valueRead = converter.Read(ref reader, typeof(T), null);
```

Comment:
It would be good to add a comment that we're explicitly trying to pass null options to make sure we don't blow up unexpectedly.